File: app/src/models/certificate.ts
export interface CertificateSubject {
  common_name?: string;
  organization?: string;
  organizational_unit?: string;
  locality?: string;
  state?: string;
  country?: string;
}

export interface CertificateInfo {
  subject: CertificateSubject;
  issuer: CertificateSubject;
  serial_number: string;
  version: number;
  not_before: string;
  not_after: string;
  subject_alternative_names: string[];
  public_key_algorithm: string;
  public_key_size?: number;
  signature_algorithm: string;
  fingerprint_sha256: string;
}

export interface CertificateChain {
  certificates: CertificateInfo[];
  is_valid: boolean;
  validation_errors: string[];
}

export interface TlsInfo {
  host: string;
  port: number;
  certificate_chain: CertificateChain;
  raw_output?: string;
}

File: src-tauri/src/models/certificate.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CertificateInfo {
    pub subject: CertificateSubject,
    pub issuer: CertificateSubject,
    pub serial_number: String,
    pub version: i32,
    pub not_before: String,
    pub not_after: String,
    pub subject_alternative_names: Vec<String>,
    pub public_key_algorithm: String,
    pub public_key_size: Option<u32>,
    pub signature_algorithm: String,
    pub fingerprint_sha256: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CertificateSubject {
    pub common_name: Option<String>,
    pub organization: Option<String>,
    pub organizational_unit: Option<String>,
    pub locality: Option<String>,
    pub state: Option<String>,
    pub country: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CertificateChain {
    pub certificates: Vec<CertificateInfo>,
    pub is_valid: bool,
    pub validation_errors: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TlsInfo {
    pub host: String,
    pub port: u16,
    pub certificate_chain: CertificateChain,
    pub raw_output: Option<String>,
}
